### DISCUSSION TITLE
Val of the same weight in different versions

### DISCUSSION THREAD

**Discussion by 347134110** [ultralytics/ultralytics#6579] [Q&A]
Setting the same config，the same weight val in version(8.0.114) would get a better mAP than val in version(8.0.202).
Do I miss anything?

**Comment by RizwanMunawar** [comment]
@347134110, appreciate you providing this information. Indeed, it's a possibility as we frequently update functions, which could result in fluctuations in Map values during validation. Just to clarify, the fundamental algorithm remains unchanged.

**Reply by 347134110** [reply]
Thank you for your reply！！

**Reply by pderrenger** [reply]
@347134110 you're welcome! If you have any more questions or need further assistance, feel free to reach out. Happy coding! 😊🚀

**Comment by 347134110** [comment]
您好，您的邮件我已经收到了，我会尽快查看，谢谢

**Reply by pderrenger** [reply]
@347134110 hello! It's not uncommon for different versions of a model to yield slightly different validation results due to changes in the codebase, such as updates to dependencies, bug fixes, or new features. When comparing mAP scores across versions, ensure that all other factors remain constant, including the dataset, configuration, and any data augmentation or preprocessing steps.

If you're observing a significant discrepancy in mAP between versions, it could be worth reviewing the release notes for any changes that might affect validation performance. Additionally, you might want to double-check that the validation dataset and settings are indeed identical for both runs.

If everything checks out and the difference in performance persists, it could be valuable to share your findings as feedback on the repository so that the team can investigate further. Your insights could help improve YOLOv8 and ensure consistent performance across versions. 🚀

For detailed guidance on validation, you can refer to the [Val documentation](https://docs.ultralytics.com/modes/val). If you have further questions or need assistance, don't hesitate to reach out!

**Comment by 347134110** [comment]
您好，您的邮件我已经收到了，我会尽快查看，谢谢

**Reply by pderrenger** [reply]
@347134110 您好！👋

关于您提到的在不同版本的YOLOv8中使用相同配置和权重进行验证但得到不同mAP的情况，这可能是由于版本之间的内部改进或者微小的代码变化导致的。我们不断地在优化和改进我们的模型和代码库，以提供更好的性能和更准确的结果。

如果您在更新版本中遇到性能下降的情况，建议检查以下几点：
1. 确认两个版本中是否使用了完全相同的验证数据集和配置文件。
2. 查看版本更新日志，了解是否有影响模型性能的重大更改。
3. 尝试在新版本中重新训练模型，以利用任何性能改进。

如果问题仍然存在，欢迎您在GitHub issue中提供更多详细信息，我们的团队会进一步帮助您解决问题。🛠️

谢谢！
